library/alloc/src/collections/btree/map/entry.rs

```diff
@@ -71,6 +71,41 @@ impl<K: Debug + Ord, V: Debug> Debug for OccupiedEntry<'_, K, V> {
     }
 }
 
+/// The error returned by [`try_insert`](BTreeMap::try_insert) when the key already exists.
+///
+/// Contains the occupied entry, and the value that was not inserted.
+#[unstable(feature = "map_try_insert", issue = "none")]
+pub struct OccupiedError<'a, K: 'a, V: 'a> {
+    /// The entry in the map that was already occupied.
+    pub entry: OccupiedEntry<'a, K, V>,
+    /// The value which was not inserted, because the entry was already occupied.
+    pub value: V,
```

Comment:
I made these fields public instead of adding accessors. This is somewhat uncommon in `std`, but we do the same for `std::ops::Range`.

Accessor functions here would get somewhat messy, because taking ownership of one *or both* of the fields should be possible. The alternative is `.entry()`, `.value()`, `.into_entry()`, `.into_value()` and `.into_entry_and_value() -> (OccupiedEntry, V)`. This can get confusing, as `.entry().key()` would work, but `.entry().remove()` would need `.into_entry().remove()` instead, which would also consume the `value`.